src/core/thread/mle_types.hpp

```diff
@@ -257,22 +257,24 @@ enum
  */
 enum
 {
-    kRegistrationDelayDefault         = 1200, ///< In seconds.
-    kMlrTimeoutDefault                = 3600, ///< In seconds.
-    kMlrTimeoutMin                    = 300,  ///< In seconds.
-    kBackboneRouterRegistrationJitter = 5,    ///< In seconds.
-    kParentAggregateDelay             = 5,    ///< In seconds.
-    kNoBufDelay                       = 5,    ///< In seconds.
-    kImmediateReRegisterDelay         = 1,    ///< In seconds.
-    KResponseTimeoutDelay             = 30,   ///< In seconds.
-    kDuaDadPeriod                     = 100,  ///< In seconds. Time period after which the address
-                                              ///< becomes "Preferred" if no duplicate address error.
+    kRegistrationDelayDefault         = 1200,              ///< In seconds.
+    kMlrTimeoutDefault                = 3600,              ///< In seconds.
+    kMlrTimeoutMin                    = 300,               ///< In seconds.
+    kMlrTimeoutMax                    = 0x7fffffff / 1000, ///< In seconds (about 24 days).
```

Comment:
Here, I limited the max MLR timeout to be about 24 days, which is not required by the Spec. 
This make expire time calculation simple, not having to cope with overflow, because the MLR Timeout in Milliseconds will not exceed the `kDistantFuture`. 

However it has two side-effects:

- Users can never set MLR timeout larger than `kMlrTimeoutMax` on a OpenThread BBR 
- OT devices that registers to BBR of another vendor would use `kMlrTimeoutMax` if the actual MLR timeout is larger than `kMlrTimeoutMax`. 

Personally I believe both side-effects are acceptable because a MLR timeout of 24 days should be more than enough for most situations. 

If MLR timeout is allowed to exceed the defined `kMlrTimeoutMax`, we can't not use `TimeMilli` for maintaining expire times because the expire time will be larger than `kDistantFuture`. We will have to use a per-second timestamp for expire time. However, currently we don't have an working implementation of per-second timer or timestamp. 

Thoughts? @jwhui @librasungirl

Reply:
I am fine with limiting MLR timeout to 24 days for now.